src/shared/cloudformation/templateRegistry.ts

```diff
@@ -21,8 +22,40 @@ export interface TemplateDatum {
 
 export class CloudFormationTemplateRegistry extends WatchedFiles<CloudFormation.Template> {
     protected name: string = 'CloudFormationTemplateRegistry'
-    protected async load(path: string): Promise<CloudFormation.Template> {
-        return await CloudFormation.load(path)
+    protected async load(path: string): Promise<CloudFormation.Template | undefined> {
+        // P0: Assume all template.yaml/yml files are CFN templates and assign correct JSON schema.
+        // P1: Alter registry functionality to search ALL YAML files and apply JSON schemas + add to registry based on validity
+
+        let template: CloudFormation.Template | undefined
+        try {
+            template = await CloudFormation.load(path)
+        } catch (e) {
+            await updateYamlSchemasArray(path, 'none')
+            return undefined
+        }
+
+        // same heuristic used by cfn-lint team:
+        // https://github.com/aws-cloudformation/aws-cfn-lint-visual-studio-code/blob/629de0bac4f36cfc6534e409a6f6766a2240992f/client/src/yaml-support/yaml-schema.ts#L39-L51
+        if (template.AWSTemplateFormatVersion || template.Resources) {
+            if (template.Transform && template.Transform.toString().startsWith('AWS::Serverless')) {
+                // apply serverless schema
+                await updateYamlSchemasArray(path, 'sam')
+            } else {
+                // apply cfn schema
+                await updateYamlSchemasArray(path, 'cfn')
+            }
+
+            return template
+        }
+
+        await updateYamlSchemasArray(path, 'none')
+        return undefined
+    }
+
+    // handles delete case
+    public async remove(path: string | vscode.Uri) {
+        await updateYamlSchemasArray(typeof path === 'string' ? path : pathutils.normalize(path.fsPath), 'none'),
```

Comment:
What's with the comma + indented super call? Looks weird to me for some reason

Reply:
I'm actually unsure, the comma may have been a typo and the indent was probably Prettier.